worldedit-core/src/main/java/com/sk89q/worldedit/extent/reorder/MultiStageReorder.java

```diff
@@ -94,58 +99,76 @@ public void setEnabled(boolean enabled) {
     }
 
     public boolean commitRequired() {
-        return stage1.size() > 0 || stage2.size() > 0 || stage3.size() > 0;
+        return enabled && stages.stream().anyMatch(stage -> stage.size() > 0);
+    }
+
+    /**
+     * Gets the stage priority of the block.
+     *
+     * @param block The block
+     * @return The priority
+     */
+    public int getPlacementPriority(BlockStateHolder block) {
+        if (Blocks.shouldPlaceLate(block.getBlockType())) {
+            return 1;
+        } else if (Blocks.shouldPlaceLast(block.getBlockType())) {
+            // Place torches, etc. last
+            return 2;
+        } else if (Blocks.shouldPlaceFinal(block.getBlockType())) {
+            // Place signs, reed, etc even later
+            return 3;
+        } else {
+            return 0;
+        }
     }
 
     @Override
     public boolean setBlock(BlockVector3 location, BlockStateHolder block) throws WorldEditException {
-        BlockState existing = getBlock(location);
-
         if (!enabled) {
             return super.setBlock(location, block);
         }
 
-        if (Blocks.shouldPlaceLast(block.getBlockType())) {
-            // Place torches, etc. last
-            stage2.add(location, block);
-            return !existing.equalsFuzzy(block);
-        } else if (Blocks.shouldPlaceFinal(block.getBlockType())) {
-            // Place signs, reed, etc even later
-            stage3.add(location, block);
-            return !existing.equalsFuzzy(block);
-        } else if (Blocks.shouldPlaceLast(existing.getBlockType())) {
+        BlockState existing = getBlock(location);
+        int priority = getPlacementPriority(block);
+        int srcPriority = getPlacementPriority(existing);
+
+        if (srcPriority == 1 || srcPriority == 2) {
```

Comment:
This is starting to feel like we should have a `RemovalPriority` as well -- this obviously doesn't mirror any placement priority, because it could be refactored to `<` or `>` in that case, but both `0` and `3` are excluded here.

Or maybe you did intend to include 3, and this should be `srcPriority > 0`.

Reply:
I've made some CLEAR levels of priority and made them get used - this fixes a few issues.